### PR TITLE
VZ-7423.  Initial build pipeline for PSR, inject worker image

### PR DESCRIPTION
**Pull request by michael-cico**
Initial Jenkins pipeline to build the `psr-backend` worker image and `psrctl` binaries and push them to Github/ObjectStore accordingly.
- Also injects the built worker image as the default image into the `psrctl` binary for now; we will likely need to revisit this and how we expect people to use it.

Build details:
- deposits CLI binraries into the per-commit buckets in Objectstore
- pushdes `psr-backend` images to Github, using `-jenkins` suffix for non-master builds

Makefile/psrctl changes:
- Flips the `DOCKER_REGISTRY` and `DOCKER_REPO` purposes, as the names didn't reflect how they were being used
- Adds a `workerImage` CLI option to override the worker image
- Injects the built worker image into the CLI as the default using `ldflags` for now, may need to revisit
- Default worker image is shown in the `psrctl` version info
- Uses the `verrazzano-development-version` for the tool/image versioning for now

To see the default worker image:
```
% psrctl version

Version:     v1.5.0
BuildDate:   2022-11-15T09:29:05Z
GitCommit:   717db24bdeb413955047cbe926970285f60dc518
WorkerImage: iad.ocir.io/odsbuilddev/mcico/dev/psr-backend:local-717db24bd
```

### CODE REVIEW

File: ci/psr/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,517 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (michael-cico) [inline ci/psr/Jenkinsfile:1]: fix copyright

File: tools/psr/psrctl/cmd/constants/constants.go [hunk 2]
```diff
@@ -23,4 +23,16 @@ const (
 	FlagScenarioDir      = "scenario-directory"
 	FlagScenarioDirShort = "d"
 	FlagScenarioDirHelp  = `a directory that contains a scenario directory at any level in the directory tree.  This allows you to run scenarios that are not compiled into the psrctl binary.`
+
```
Comment (michael-cico) [inline tools/psr/psrctl/cmd/constants/constants.go:26]: add copyright

File: tools/psr/manifests/charts/worker/values.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 imagePullSecrets: [ ]
 
-imageName: ghcr.io/verrazzano/psr-backend:local-c10faf9
+#imageName: ghcr.io/verrazzano/psr-backend:local-c10faf9
```
Comment (mbenmclean) [inline tools/psr/manifests/charts/worker/values.yaml:6]: delete duplicate commented line

**Review by stevshap — APPROVED** [review]

File: ci/psr/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+            image "${RUNNER_DOCKER_IMAGE}"
+            args "${RUNNER_DOCKER_ARGS}"
+            registryUrl "${RUNNER_DOCKER_REGISTRY_URL}"
+            registryCredentialsId 'ocir-pull-and-push-account'
+            label "${agentLabel}"
+        }
+    }
+
+    parameters {
+        booleanParam (description: 'Whether to perform a scan of the built images', name: 'PERFORM_SCAN', defaultValue: false)
```
Comment (mbenmclean) [inline ci/psr/Jenkinsfile:30]: Do we even want to be scanning our backend image. It's just internal

File: ci/psr/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+                    post {
+                        always {
+                            sh """
+                            cd ${PSR_PATH}
+                            #cp coverage.html ${WORKSPACE}
+                            #cp coverage.xml ${WORKSPACE}
+                            ${GO_REPO_PATH}/build/copy-junit-output.sh ${WORKSPACE}
+                             """
+                            junit testResults: '**/*test-result.xml', allowEmptyResults: true
+                            /*cobertura(coberturaReportFile: 'coverage.xml',
```
Comment (mbenmclean) [inline ci/psr/Jenkinsfile:196]: remove comments

**Review by mbenmclean — APPROVED** [review]

File: tools/psr/manifests/charts/worker/values.yaml (hunk 3 above)
Comment (michael-cico) [reply tools/psr/manifests/charts/worker/values.yaml:6]: It's there as a reference for now, so folks know what the key is.

File: ci/psr/Jenkinsfile (hunk 5 above)
Comment (michael-cico) [reply ci/psr/Jenkinsfile:196]: I left it intentionally because I think we want code coverage numbers from talking to Paul.

**Pull request merged**